devicetypes/technisat/technisat-series-switch.src/technisat-series-switch.groovy

```diff
@@ -0,0 +1,439 @@
+	def ccObj = version ? zwave.commandClass(cmd.commandClass, version) : zwave.commandClass(cmd.commandClass)
+	def encapsulatedCommand = ccObj?.command(cmd.command)?.parse(cmd.data)
+	if (encapsulatedCommand) {
+		log.debug "Parsed Crc16Encap into: ${encapsulatedCommand}"
+		zwaveEvent(encapsulatedCommand)
+	} else {
+		log.warn "Unable to extract CRC16 command from $cmd"
+	}
+}
+
+private secEncap(physicalgraph.zwave.Command cmd) {
+	log.debug "encapsulating command using Secure Encapsulation, command: $cmd"
+	zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+}
+
+private crcEncap(physicalgraph.zwave.Command cmd) {
+	log.debug "encapsulating command using CRC16 Encapsulation, command: $cmd"
+	zwave.crc16EncapV1.crc16Encap().encapsulate(cmd).format()
+}
+
+def encapEp(endpointNumber, cmd) {
+	if (cmd instanceof physicalgraph.zwave.Command) {
+		encap(zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint: endpointNumber).encapsulate(cmd))
+	} else if (cmd.startsWith("delay")) {
+		cmd
+	} else {
+		def header = "600D00"
+		String.format("%s%02X%s", header, endpointNumber, cmd)
+	}
+}
+
+private encap(physicalgraph.zwave.Command cmd) {
+	if (zwaveInfo?.zw?.contains("s")) {
+		secEncap(cmd)
+	} else if (zwaveInfo?.cc?.contains("56")) {
+		crcEncap(cmd)
+	} else {
+		log.debug "no encapsulation supported for command: $cmd"
+		cmd.format()
+	}
+}
+
+private isConfigChanged(parameter) {
+	def settingsValue = settings."$parameter.key"
+	log.debug "isConfigChanged parameter:${parameter.key}: ${settingsValue}"
+	if (parameter.enableSwitch) {
+		if (settings."$parameter.enableKey" != null) {
+			if (settings."$parameter.enableKey" == false) {
+				settingsValue = 0;
+			}
+		}
+	}
+	if (settingsValue != null) {
+		Integer value = 0
+		if (parameter.type == "number") {
+			value = settingsValue
+		} else {
+			value = Integer.parseInt(settingsValue)
+		}
+		if (state.currentConfig."$parameter.key".value != value) {
+			state.currentConfig."$parameter.key".newValue = value
+			log.debug "${parameter.key} set:${value} value:${state.currentConfig."$parameter.key".value} newValue:${state.currentConfig."$parameter.key".newValue}"
+			return true
+		} else if (state.currentConfig."$parameter.key".status != "sync") {
+			log.debug "${parameter.key} retry to set; is:${state.currentConfig."$parameter.key".value} should:${state.currentConfig."$parameter.key".newValue}"
+			return true
+		}
+		return false
+	} else {
+		log.debug "pref value not set yet"
+		return false
+	}
+}
+
+private syncConfig() {
+	def commands = []
+	parameterMap.each {
+		if (isConfigChanged(it)) {
+			log.debug "Parameter ${it.key} has been updated from value: ${state.currentConfig."$it.key".value} to ${state.currentConfig."$it.key".newValue}"
+			state.currentConfig."$it.key".status = "syncPending"
+			commands << response(encap(zwave.configurationV2.configurationSet(configurationValue: intToParam(state.currentConfig."$it.key".newValue, it.paramZwaveSize),
```

Comment:
you should just be able to pass the value to `scaledConfigurationValue` instead of converting yourself and using `configurationValue`